papermill/tests/test_iorw.py

```diff
@@ -7,7 +7,71 @@
     from mock import Mock, patch
 
 from ..exceptions import PapermillException
-from ..iorw import HttpHandler
+from ..iorw import HttpHandler, PapermillIO
+
+
+class TestPapermillIO(unittest.TestCase):
+    """
+    Tests for `PapermillIO`.
+    """
+
+    class FakeHandler(object):
+        def __init__(self, ver):
+            self.ver = ver
+
+        def read(self, path):
+            return "contents from {} for version {}".format(path, self.ver)
+
+        def listdir(self, path):
+            return ["fake", "contents"]
+
+        def write(self, buf, path):
+            return "wrote {}".format(buf)
+
+        def pretty_path(self, path):
+            return "{}/pretty/{}".format(path, self.ver)
+
+
+    def setUp(self):
+        self.papermill_io = PapermillIO()
+        self.fake1 = self.FakeHandler(1)
+        self.fake2 = self.FakeHandler(2)
+        self.papermill_io.register("fake", self.fake1)
+
+    def test_get_handler(self):
+        self.assertEqual(self.papermill_io.get_handler("fake"), self.fake1)
+
+    def test_get_local_handler(self):
+        with self.assertRaises(PapermillException):
+            self.papermill_io.get_handler("dne")
+
+        self.papermill_io.register("local", self.fake2)
```

Comment:
the idea is that a `local` handler should be the fallback if nothing else matches?

Reply:
Yes. This was the behavior before, I just preserved it.